src/util.js

```diff
@@ -0,0 +1,44 @@
+import axios from 'axios'
+
+export function getCodeVerifier() {
+  const lowerCase = 'abcdefghijklmnopqrstuvwxyz'
+  const upperCase = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ'
+  const numbers = '1234567890'
+  const specialCharacters = "-._~"
+  const characters = lowerCase + upperCase + numbers + specialCharacters
+
+  const minLength = 43
+  const maxLength = 128
+  const stringLength = Math.floor(Math.random() * (maxLength - minLength + 1)) + minLength
+
+  let string = ''
+  for (var i = 0; i < stringLength; i++) {
+    const randomIndex = Math.floor(Math.random() * characters.length)
+    string += characters.charAt(randomIndex)
+  }
+
+  return string;
+}
+
+export function getAuthUrl() {
+    const codeVerifier = window.localStorage.getItem('DW-CODE-VERIFIER')
+    const base = 'https://data.world/oauth/authorize?'
+    const clientId= `client_id=${process.env.REACT_APP_OAUTH_CLIENT_ID}&`
+    const redirectUri = `redirect_uri=${process.env.REACT_APP_OAUTH_REDIRECT_URI}&`
+    const responseType = 'response_type=code&'
+    const codeChallengeMethod = 'code_challenge_method=plain&'
+    const codeChallenge = `code_challenge=${codeVerifier}`
+
+    return `${base}${clientId}${redirectUri}${responseType}${codeChallengeMethod}${codeChallenge}`
```

Comment:
This is a little difficult to read, why not put the pieces that are static into the template?

Reply:
Was avoiding making the line more than 100 characters long. The static parts are now in the template.